### DISCUSSION TITLE
ModuleNotFoundError: No module named 'langdetect' (e27e0c48)

### DISCUSSION THREAD

**Discussion by mistersixt** [searx/searx#2687] [Installation issue: custom]
Hi all,

I just did an "git pull origin master" + "manage.sh update_packages" on Debian Buster, ending up with this error:

Traceback (most recent call last):
  File "/usr/local/searx/searx/webapp.py", line 70, in <module>
    from searx.webadapter import get_search_query_from_webapp, get_selected_categories
  File "/usr/local/searx/searx/webadapter.py", line 5, in <module>
    from searx.query import RawTextQuery
  File "/usr/local/searx/searx/query.py", line 9, in <module>
    from searx.search import EngineRef
  File "/usr/local/searx/searx/search/__init__.py", line 33, in <module>
    from searx.search.checker import initialize as initialize_checker
  File "/usr/local/searx/searx/search/checker/__init__.py", line 3, in <module>
    from .impl import Checker
  File "/usr/local/searx/searx/search/checker/impl.py", line 12, in <module>
    from langdetect import detect_langs
ModuleNotFoundError: No module named 'langdetect'
Fri Mar 26 09:20:05 2021 - unable to load app 0 (mountpoint='') (callable not found or import error)
Fri Mar 26 09:20:05 2021 - *** no app loaded. going in full dynamic mode ***

Found this on my box ...

"/usr/local/searx/.local/lib/python3.7/site-packages/langdetect/"

... as well as:

root@mail:/var/log/uwsgi/app# dpkg -l | grep langd
ii  python3-langdetect               1.0.7-3                               all          Python library for language-detection (Python 3)
root@mail:/var/log/uwsgi/app#

Any help is welcome,

Kind regards, Joerg.

**Comment by dalf** [comment]
What is the content of uwsgi.ini ?

---
RIP issue template.

**Reply by mistersixt** [reply]
```
root@mail:/var/log/uwsgi/app# cat /usr/local/searx/dockerfiles/uwsgi.ini
[uwsgi]
# Who will run the code
uid = searx
gid = searx

# Number of workers (usually CPU count)
workers = 4

# The right granted on the created socket
chmod-socket = 666

# Plugin to use and interpretor config
single-interpreter = true
master = true
plugin = python3
lazy-apps = true
enable-threads = true

# Module to import
module = searx.webapp

# Virtualenv and python path
pythonpath = /usr/local/searx/
chdir = /usr/local/searx/searx/

# Disable logging for privacy
disable-logging=True

# But keep errors for 2 days
touch-logrotate = /run/uwsgi-logrotate
unique-cron = 15 0 -1 -1 -1 { touch /run/uwsgi-logrotate  }
log-backupname = /var/log/uwsgi/uwsgi.log.1
logto = /var/log/uwsgi/uwsgi.log

# No keep alive
# See https://github.com/searx/searx-docker/issues/24
add-header = Connection: close

# uwsgi serves the static files
# expires set to one day as Flask does
static-map = /static=/usr/local/searx/searx/static
static-expires = /* 864000
static-gzip-all = True
offload-threads = %k

# Cache
cache2 = name=searxcache,items=2000,blocks=2000,blocksize=4096,bitmap=1
root@mail:/var/log/uwsgi/app#
```

**Reply by dalf** [reply]
How did you install Searx (`git clone` I guess)? 
But more importantly, how did you install uwsgi?

**Reply by mistersixt** [reply]
Exactly, "git clone" (I followed this "Step by step installation" from the webpage initially), and uwsgi via apt/dpkg, so from the regular Debian Buster repository.

It was working fine for more than a year, with updating things with "git pull...", like once a month or something, but todays latest "git pull origin master" broke things...

**Reply by dalf** [reply]
I don't think `/usr/local/searx/dockerfiles/uwsgi.ini` is the actual uwsgi.ini (according to your description, I may be wrong).
Most probably, you should look in `/etc/uwsgi` and `/etc/default/uwsgi`.

In the `uwsgi.ini`, you should look for the values of `virtualenv` and `pythonpath`.
If there is `virtualenv=`, you should install the new dependency inside this virtualenv.

**Reply by mistersixt** [reply]
I think I know what it is, you pretty sure wanted to see the content of "/etc/uwsgi/apps-enabled/searx.ini", this is indeed quite outdated when comparing to the current "default" entries.

I think I am going to install things from scratch... thanks for all your hints and help!!!

Kind regards, Joerg.
